src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,214 @@
+            tasks.addTask(A);
+            Duke.index++;
+            String message = "    ----------------------------\n"+
+                    "Got it, I've added this task: \n"
+                    + A.toString() + "\n"
+                    + "Now you have " + Duke.index + " tasks in the list\n"
+                    +"    ----------------------------";
+            System.out.println(message);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Oops! The description of a todo cannot be empty!");
+        }
+    }
+
+    /**
+     * Executes when Duke receives an event command
+     * Adds an event task to the list of tasks
+     */
+    public void event_execute() {
+        try {
+            int i = userInput.indexOf("/");
+            String description = userInput.substring(6,i-1);
+            //String time = userInput.substring(i+1);
+            String[] split = userInput.split("at");
+            String date = split[1].substring(1);
+            LocalDate d = LocalDate.parse(date);
+            String formattedTime = d.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+            Task A = new Event(description, formattedTime);
+            tasks.addTask(A);
+            Duke.index++;
+            String message = "    ----------------------------\n"
+                    +"Got it, I've added this task: \n"
+                    + A.toString() + "\n"
+                    + "Now you have " + Duke.index + " tasks in the list\n"
+                    +"    ----------------------------";
+            System.out.println(message);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Oops! The description of an event cannot be empty and must contain a time!");
+        }
+    }
+
+    /**
+     * Executes when Duke receives a deadline command
+     * Adds a deadline task to the list of tasks
+     */
+    public void deadline_execute() {
+        try {
+            int i = userInput.indexOf("/");
+            String description = userInput.substring(9,i-1);
+            //String time = userInput.substring(i+1);
+            String[] split = userInput.split("by");
+            String date = split[1].substring(1);
+            LocalDate d = LocalDate.parse(date);
+            String formattedTime = d.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+            Task A = new Deadlines(description, formattedTime);
+            tasks.addTask(A);
+            Duke.index++;
+            String message = "    ----------------------------\n"
+                    + "Got it, I've added this task: \n"
+                    + A.toString() + "\n"
+                    + "Now you have " + Duke.index + " tasks in the list\n"
+                    + "    ----------------------------";
+            System.out.println(message);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Oops! The description of a deadline cannot be empty and must contain a time!");
+        }
+    }
```

Comment:
Since Todo, Deadline, and Event all inherit from Task, perhaps it would be better to make this an abstract method in the Task class and implement them in the individual sub classes instead?

Reply:
Or you could just have the method in the Task class and override the method in each subclass.